packages/slate-commons/src/commands/isCursorOnEdgeOfContainer.ts

```diff
@@ -0,0 +1,40 @@
+import { Editor } from 'slate';
+import type { Range, Point } from 'slate';
+import { ReactEditor } from 'slate-react';
+
+export type ContainerEdge = 'top' | 'bottom';
+
+export function isCursorOnEdgeOfContainer(
+    editor: ReactEditor,
+    container: Point,
+    child: Point,
+    edge: ContainerEdge,
+) {
+    const a = getPointRect(editor, container);
+    const b = getPointRect(editor, child);
+
+    switch (edge) {
+        case 'top':
+            return a.at(0)?.top === b.at(0)?.top;
+        case 'bottom':
+            return a.at(-1)?.bottom === b.at(-1)?.bottom;
+    }
+}
+
+function getPointRect(editor: ReactEditor, point: Point) {
+    const range = Editor.range(editor, { ...point, offset: Math.max(point.offset, 0) });
+    return getRangeRect(editor, range);
+}
+
+function getRangeRect(editor: ReactEditor, range: Range) {
+    const domRange = ReactEditor.toDOMRange(editor, range);
```

Comment:
This can throw an error if ReactEditor cannot match range to DOM nodes. 

See #260